File: engine/tests/fixtures/canonical.js
function solve(values, limit) {
  const seed = [1, 2, 3];
  const seen = new Map();
  const unique = new Set(values);
  const prefix = values.slice(0, limit);
  for (const value of prefix) {
    if (value > 0) seen.set(value, solve(prefix, limit - 1));
  }
  while (limit > 0) limit -= 1;
  return seen;
}

File: engine/tests/fixtures/canonical.rs
fn solve(values: Vec<i64>, mut limit: usize) -> std::collections::HashMap<i64, usize> {
    let seed = [1, 2, 3];
    let mut seen = std::collections::HashMap::new();
    let unique = std::collections::HashSet::from_iter(values.clone());
    let prefix = values[..limit].to_vec();
    for value in &prefix {
        if *value > 0 {
            seen.insert(*value, solve(prefix.clone(), limit - 1).len());
        }
    }
    while limit > 0 {
        limit -= 1;
    }
    seen
}
